src/plonk_recursion.rs

```diff
@@ -194,9 +294,31 @@ pub fn recursive_verification_circuit<
         builder.copy(public_inputs.halo_u_l[i].routable_target(), u_l[i]);
         builder.copy(public_inputs.halo_u_r[i].routable_target(), u_r[i]);
     }
+    let shift = 2 + degree_pow;
+    for i in 0..num_old_proofs {
+        builder.copy(
+            old_proofs[i].halo_g.x,
```

Comment:
Hm if I'm reading this correctly, it seems like it's copying "inner old proof" data to "outer old proof" data?

I think we should copy `proof.halo_g` into and `ipa_challenges` into `public_inputs.old_proofs`. Although it's a bit awkward since we have one each of those variables for now, but long term we should have one `halo_g` and `ipa_challenges` for each inner proof, and the same number of "old proof outputs", if I understand correctly.

Reply:
What do you mean by "outer old proof" data?
What I'm trying to do here is just continue copying the public inputs to the virtual targets, similar to what is done from lines 255-296, e.g.,
```rust
builder.copy(
        public_inputs.o_plonk_z_right.routable_target(),
        proof.o_right.o_plonk_z,
    );
```

Reply:
Hm I think of those other copies as "outputting" bits of the inner proof that still need to be checked. If I understand correctly, the `g` points in `old_proofs` have been fully processed by this point, since they're incorporated into this circuit's opening argument by `verify_old_proof_evaluation`. It seems like `proof.halo_g` is the one that still needs to be checked by the next proof. Is that right?

Reply:
Ok I see what you mean. I'll make the changes in another PR.